### PR TITLE
[4.0] Replace com_content 'Image Float' with 'Image Class

### PR DESCRIPTION
**Pull request by ciar4n**
This is a rebase of #17402. Opened on request https://github.com/joomla/joomla-cms/issues/30924#issuecomment-705354364

Pull Request for Issue #17399 #30924 .

### Summary of Changes
Replaces the 'Image Float ' field in com_contents with an 'Image Class' field

### Testing Instructions
Navigate to the **Images and Links** tab in article edit. Set an image and add a utility class to the **Image Class** field . Check frontend and ensure class has been applied correctly.

### CODE REVIEW

**Comment by brianteeman** [comment]
Looks a good proposal and I see that you have considered the legacy issues. Thanks

File: templates/cassiopeia/scss/blocks/_legacy.scss [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Legacy
+
+.left.item-image {
+  float: left;
+}
+.right.item-image {
+  float: right;
+}
```
Comment (richard67) [inline templates/cassiopeia/scss/blocks/_legacy.scss]:
```suggestion
}

```
Fixes SCSS CS.

Comment (ciar4n) [reply templates/cassiopeia/scss/blocks/_legacy.scss]: Thank you @richard67

File: administrator/language/en-GB/com_content.ini [hunk 2]
```diff
@@ -94,9 +94,9 @@ COM_CONTENT_FILTER_FEATURED_NO="Unfeatured Articles"
 COM_CONTENT_FILTER_FEATURED_YES="Featured Articles"
 COM_CONTENT_FILTER_SEARCH_DESC="Search in title, alias and note. Prefix with ID: or AUTHOR: or CONTENT: to search for an article ID, article author or search in article content."
 COM_CONTENT_FILTER_SEARCH_LABEL="Search Articles"
-COM_CONTENT_FLOAT_FULLTEXT_LABEL="Full Text Image Float"
-COM_CONTENT_FLOAT_INTRO_LABEL="Intro Image Float"
-COM_CONTENT_FLOAT_LABEL="Image Float"
+COM_CONTENT_IMAGE_INTRO_CLASS_LABEL="Intro Image Class"
```
Comment (Quy) [inline administrator/language/en-GB/com_content.ini]: Please sort keys in alpha order.

**Comment by adj9** [comment]
With the inspection of HTML I see the field
Schermata 2020-10-10 alle 12 22 08

The image of the test is the logo Joomla!

The Image Class field is perhaps more like a list of elements <left center right> than text.

**Comment by adj9** [comment]
I have tested this item :white_check_mark: successfully on 65f5d4b08c56dc879ab02fec19523afb82d006ca<br /><br />Done :) <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31017">issues.joomla.org/tracker/joomla-cms/31017</a>.</sub>

File: administrator/language/en-GB/com_content.ini [hunk 3]
```diff
@@ -94,15 +95,14 @@ COM_CONTENT_FILTER_FEATURED_NO="Unfeatured Articles"
...
-COM_CONTENT_IMAGE_FULLTEXT_CLASS_LABEL="Full Text Image Class"
-COM_CONTENT_FIELD_IMAGE_CLASS_LABEL="Image Class"
 COM_CONTENT_HEADING_ASSOCIATION="Association"
 COM_CONTENT_HEADING_DATE_CREATED="Date Created"
 COM_CONTENT_HEADING_DATE_MODIFIED="Date Modified"
 COM_CONTENT_HEADING_DATE_PUBLISH_DOWN="Finish Publishing"
 COM_CONTENT_HEADING_DATE_PUBLISH_UP="Start Publishing"
 COM_CONTENT_ID_LABEL="ID"
+COM_CONTENT_IMAGE_INTRO_CLASS_LABEL="Intro Image Class"
+COM_CONTENT_IMAGE_FULLTEXT_CLASS_LABEL="Full Text Image Class"
```
Comment (infograf768) [inline administrator/language/en-GB/com_content.ini:104]: FULLTEXT alpha order comes before INTRO

**Comment by richard67** [comment]
The previous test is still valid because the commit after that has only changed order of the language string. I've added back the test result in the issue tracker.

File: administrator/language/en-GB/com_content.ini (hunk 3 above)
Comment (richard67) [reply administrator/language/en-GB/com_content.ini:104]: Oh, seems I was too fast with adding back the test result. @ciar4n Could you just fix this, too? I'll add back the test again when done.

**Comment by Quy** [comment]
Please update the following and remove obsolete strings:

https://github.com/ciar4n/joomla-cms/blob/5a5082158b8b0701384bcd842f2027a309bdeb51/components/com_content/forms/article.xml#L250

https://github.com/ciar4n/joomla-cms/blob/5a5082158b8b0701384bcd842f2027a309bdeb51/components/com_content/forms/article.xml#L283

**Comment by ChrisHoefliger** [comment]
Tested successfully, class is added.

**Comment by richard67** [comment]
It seems the PR meanwhile has a conflict in file `layouts/joomla/content/intro_image.php`.

**Comment by richard67** [comment]
I've resolved the conflict.

@ChrisHoefliger Could you mark your test result in the issue tracker here? https://issues.joomla.org/tracker/joomla-cms/31017 . Just use the "Test this" button, then select the appropriate test result and then submitt. Please do it in the same way with the other pull requests which you have tested today and where you haven't dome this yet. Thanks in advance.

**Comment by nadjak77** [comment]
I have tested this item :white_check_mark: successfully on 1185c10f641997bd5bf60a63792e1a2319aeeab1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31017">issues.joomla.org/tracker/joomla-cms/31017</a>.</sub>

**Comment by richard67** [comment]
It still needs to alpha sort the language strings as mentioned by @infograf768 above.

**Comment by BiancaOHSN** [comment]
I have tested this item :white_check_mark: successfully on 1185c10f641997bd5bf60a63792e1a2319aeeab1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31017">issues.joomla.org/tracker/joomla-cms/31017</a>.</sub>

File: administrator/language/en-GB/com_content.ini (hunk 3 above)
Comment (richard67) [reply administrator/language/en-GB/com_content.ini:104]: @ciar4n I've allowed myself to fix that one for you.

**Comment by richard67** [comment]
I've allowed myself to fix the language string ordering and I've restored the previous test results.

**Comment by richard67** [comment]
Can be set to RTC as soon as automated tests have passed.

**Comment by nadjak77** [comment]
I have tested this item :white_check_mark: successfully on 8374b20aec0df220a6f75563b85eccce67c518aa<br /><br />testes successfully again :)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31017">issues.joomla.org/tracker/joomla-cms/31017</a>.</sub>

**Comment by richard67** [comment]
@ciar4n Will this PR require documentation changes when being merged? If so, let me know and I set the corresponding label. If you could add information about that in the description in that case would be great, too. Thanks in advance for feedback.

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31017">issues.joomla.org/tracker/joomla-cms/31017</a>.</sub>

**Comment by drmenzelit** [comment]
Thanks

**Pull request merged**